### ISSUE TITLE
postgresql: create doesn't support returning

### ISSUE THREAD

**Issue by james-lawrence** [coleifer/peewee#1831]
was attempting to insert records into the database and ran into this issue.

problem is two fold: 
- create doesn't support returning afaik.
- documentation around returning is [rather sparse](http://docs.peewee-orm.com/en/latest/peewee/querying.html#returning-clause).

```python
m = Example.create(title = details.title).returning(Example)
# INSERT INTO "example" ("title") VALUES (%s) RETURNING "example"."id"
# *** AttributeError: 'KbImports' object has no attribute 'returning'
```

```
m = Example.insert(title = details.title).returning(Example).execute()
x = [y for y in m]
x[0].id
# INSERT INTO "example" ("title") VALUES (%s) RETURNING "example"."id", "kb_imports"."created_at", "kb_imports"."title", "kb_imports"."updated_at"
# *** AttributeError: 'tuple' object has no attribute 'id'
```

how does one go from an insert to a model object....

**Comment by james-lawrence** [comment]
I've discovered how to do it.... mainly by happenstance. but my original two issues still hold.

```
ins = Example.insert(title = details.title).returning(Example).objects(Example).execute()
```

**Comment by coleifer** [comment]
Correct. `create()` is not intended to be used with `returning()` since the return value of `create()` is already a model instance (with any auto-incrementing ID populated).

`returning()` can be used with `insert()` (or `insert_many`). By default it returns tuples of values, but as you noted, it is possible to return model instances.

**Comment by james-lawrence** [comment]
@coleifer documentation should be improved for returning at a minimum before this is closed. took far too long to discover how to get general (and expected) response from peewee when using returning.

**Comment by coleifer** [comment]
What specifically did you find confusing?

**Comment by james-lawrence** [comment]
specifically [returning clause docs](http://docs.peewee-orm.com/en/latest/peewee/querying.html#returning-clause)

make no reference to the ability to adjust the results of the query. an explicit link back to the [previous section](http://docs.peewee-orm.com/en/latest/peewee/querying.html#retrieving-row-tuples-dictionaries-namedtuples) would be very helpful. as when I was looking into the returning clause I never saw the ability to adjust the deserialization section.

which is pretty much the first thing someone is going to want to do after looking into returning().

**Comment by coleifer** [comment]
Thanks, I noticed a quirk when adding tests... UPDATE and DELETE will return `Model` instances by default when you have a `RETURNING` clause. I think there was a regression that caused `INSERT` to return `tuple` objects by default (when you have a non-default returning clause).

I've also addressed this inconsistency, so that now if you specify a RETURNING clause on an INSERT, the default row type returned will be model instances.

I've also updated the docs.
